client/docs/modules.md

```diff
@@ -0,0 +1,39 @@
+# Monorepo: Modules
+
+Modules are the packages that can be found in the `modules/` folder. They're intended for pure ESM output, and are compiled rapidly using Estrella, an esbuild-based utility. They have an important file structure (don't change it), which looks like the following:
+
+```
+module:
+- src/
+  - index.ts
+  - **/*.ts
+- tests/
+  - *.ts
+- package.json
+- tsconfig.json
+```
+
+The `src/` folder is compiled into a `dist/` folder when the module is built. Similarly, files in the `tests/` folder are compiled into the `tests/dist/` folder when the tests are built.
+
+Other than the inheritance of development dependencies and tooling configurations, these modules act like fairly normal NPM packages. If you need to add a dependency to one, you can either navigate to the module and do a normal `npm i <dep>` command, or manually add it to the `package.json`. After you do that, you need to make sure to run the `npm run bootstrap` command at the root. This ensures that all the fragile symlinks get restored by Lerna.
+
+To improve performance of the monorepo, building and watching of modules isn't done per module, but instead globally at root. It is recommended that you just build the entire monorepo rather than trying to build individual packages. This is not an insane recommendation because module building is 100% incremental. If a module has not changed, it won't get rebuilt.
+
+There is a template for making modules, and it can be found in the `templates/module-template` folder.
+
+Here is how to use that template:
+
+1. Create a folder in `modules/` that is the name of your package.
+  It isn't strictly required that it is the name of your package, but it gets confusing otherwise.
+  If you want to build the module as a Node package instead of for the browser, prefix the folder name with `node-`.
+
+2. Copy the contents of `templates/module-template` into the new folder.
+
+3. Edit `package.json`.
+  There is a few properties you should edit. They are: `name`, `description`, and `version`. Everything else is fine being left alone.
```

Comment:
```suggestion
  There are a few properties you should edit. They are: `name`, `description`, and `version`. Everything else is fine being left alone.
```

Reply:
Any recommendations on sub-package versioning? I imagine it would probably be easiest to use [calver](https://calver.org/) for all of them, since "1.0" isn't really meaningful and they're entirely dependent on being in the Wikijump ecosystem.

Reply:
Lerna is in independent mode, which means packages get versioned arbitrarily depending on whether or not they have changed since they've last been published. I left versioning entirely alone in this because it's a bit of a can of worms. If you don't mind, I'd like to handle that in a separate PR when we start actually uploading packages.

Reply:
If they're bumped automatically, what does that mean exactly? Does it just change the minor version every time or something?

This doesn't seem like an argument against calver per se, but I also don't really get what lerna does with versions so.

Reply:
To be honest, I haven't played with Lerna much. iirc, you give it `major | minor | patch` and it bumps? I have to look into it, and calver seems appealing to me.